### ISSUE TITLE
[Help] PyLS crashes Neovim immediately

### ISSUE THREAD

**Issue by Priddles** [autozimu/LanguageClient-neovim#661]
## Describe the problem
Almost as soon as PyLS starts, Neovim crashes.

## Environment
- OS: Windows 10
- Neovim version: `0.3.1`
- This plugin version: `da723a0`
- This plugin's binary version: `0.1.129`
- Minimal vimrc content: bottom of this [gist](https://gist.github.com/Priddles/7bce80d2d578505d8652085261fa9a60).
- Language server link and version: [python-language-server[rope,pyflakes]](https://github.com/palantir/python-language-server) `0.21.2`

[min-vimrc.vim]: https://github.com/autozimu/LanguageClient-neovim/blob/next/min-vimrc.vim

## To Reproduce
Open any buffer with extension `.py`:
1. Run `nvim -u min-init.vim does-not-exist.py`
1. Crash.
1. Run `nvim -u min-init.vim actually-exists.py`
1. Crash again.

## Additional context
See [logs](https://gist.github.com/Priddles/7bce80d2d578505d8652085261fa9a60).

**Comment by KillTheMule** [comment]
If this is really neovim crashing, that would (at least also) be a bug in neovim. An issue on the neovim repo would be appreciated, in particular if you can get us a [backtrace](https://github.com/neovim/neovim/wiki/Development-tips#debugging), and the [neovim log file](https://github.com/neovim/neovim/wiki/Development-tips#log-file-location).

(e) Oh heck, windows, so I guess no backtrace. Not sure what there's to do, an issue would still be good.

**Comment by Priddles** [comment]
I've got [vim-lsp](https://github.com/prabirshrestha/vim-lsp) working fine (somehow).

The Neovim repo's issue template asks that you reproduce the problem using `-u NORC`. Would submitting this be okay?

Also, I don't know if anyone else has got Windows + Neovim + LC-n + PyLS working so I think I'll hold back on submitting an issue until I know that another person has had a similar problem.

**Comment by autozimu** [comment]
I'd suggest filing an issue on neovim side anyway.

In any circumstance, neovim should not crash, no matter what this plugin does or crashes.

**Comment by KillTheMule** [comment]
> The Neovim repo's issue template asks that you reproduce the problem using `-u NORC`. Would submitting this be okay?

Definitely, your reproduction steps are very good, and as autozimu say, neovim shouldn't crash under any circumstances.

**Comment by Priddles** [comment]
Neovim didn't crash when I built it using the Mingw64 debug configuration!

Details [here](https://github.com/neovim/neovim/issues/9266#issuecomment-441150313).

**Comment by erw7** [comment]
@Priddles You need to use `pyls` 0.21.3 or later on Windows. Prior `pyls` had the problem of terminating the parent process on Windows. Still it is strange that `nvim.exe`, the parent process of `languageclient.exe`, will terminate.

Ref. palantir/python-language-server#434

**Comment by autozimu** [comment]
Can you try out the suggestions above?

**Comment by Priddles** [comment]
@erw7 @autozimu 
I was sure I was using the latest `pyls` at the time I opened the issue, and I had been using that version with no problems to this day with [vim-lsp](https://github.com/prabirshrestha/vim-lsp).

I tried the new `pyls 0.21.3` and *LanguageClient* did not crash and by all indication appeared to be in perfect working order.

Despite `pyls 0.21.2` working fine with *vim-lsp* and not *LanguageClient*, it seems the root cause was a bug in `pyls` itself. As such, I'll be closing the issue.

Thanks for everyone's help!
